<launch>
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="name" default="_uvdar" />
  <arg name="portname" default="/dev/MRS_MODULE1" />
  <arg name="profiler" default="$(optenv PROFILER false)" />

  <arg name="sequence_file" default="$(find uvdar_core)/config/selected.txt"/>

  <group ns="$(arg UAV_NAME)">

    <!-- launch the node -->
    <node pkg="nodelet" type="nodelet" name="serial_uvdar" args="standalone baca_protocol/BacaProtocol" respawn="true" output="screen">
      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>

      <rosparam file="$(find mrs_serial)/config/mrs_serial.yaml" />

      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="portname" value="$(arg portname)"/>
      <param name="use_timeout" value="true"/>

      <!-- Publishers -->
      <remap from="~range" to="/$(arg UAV_NAME)/garmin/range" />
      <remap from="~range_up" to="/$(arg UAV_NAME)/garmin/range_up" />
      <remap from="~profiler" to="profiler" />
      <remap from="~baca_protocol_out" to="~received_message" />

        <!-- Subscribers -->
      <remap from="~baca_protocol_in" to="~send_message" />
      <remap from="~raw_in" to="~send_raw_message" />

    </node>

    <node name="uvdar_led_manager_node" pkg="uvdar_core" type="uvdar_led_manager_node" output="screen" respawn="true">
      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>
      <param name="sequence_file" type="string" value="$(arg sequence_file)" />
        <!-- Publisher -->
      <remap from="~baca_protocol_out" to="/$(arg UAV_NAME)/serial_uvdar/send_message" />

    </node>

  </group>

</launch>
